CMakeLists.txt

```diff
@@ -0,0 +1,150 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+cmake_minimum_required(VERSION 3.21...3.24)
+
+# LLVM requires CMP0116 for tblgen: https://reviews.llvm.org/D101083
+# CMP0116: Ninja generators transform `DEPFILE`s from `add_custom_command()`
+# New in CMake 3.20. https://cmake.org/cmake/help/latest/policy/CMP0116.html
+set(CMAKE_POLICY_DEFAULT_CMP0116 OLD)
+if(POLICY CMP0116)
+  cmake_policy(SET CMP0116 OLD)
+endif()
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+project(LIGHTHOUSE C CXX)
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
+
+#-------------------------------------------------------------------------------
+# Input Dialects
+#-------------------------------------------------------------------------------
+
+option(LIGHTHOUSE_INPUT_TORCH "Builds support for compiling Torch MLIR programs" ON)
+
+#-------------------------------------------------------------------------------
+# Python
+# If building features that require Python development, find them early in
+# one invocation (some CMake versions are sensitive to resolving out of order).
+#-------------------------------------------------------------------------------
+option(LIGHTHOUSE_BUILD_PYTHON_BINDINGS "Builds the Lighthouse python bindings" OFF)
```

Comment:
I'd leave this ON for now, to make it easier to test.

Reply:
Done